cli/Makefile

```diff
@@ -108,39 +108,38 @@ ABI_UNDERSCORE :=
 endif
 EXPORTED_FUNCS := $(shell echo -e "#include \"jl_exported_funcs.inc\"\n#define XX(x) $(ABI_UNDERSCORE)x\nJL_EXPORTED_FUNCS(XX)" | $(CPP) -I$(JULIAHOME)/src - | tail -n 1)
 STRIP_EXPORTED_FUNCS := $(patsubst %,--strip-symbol=%,$(EXPORTED_FUNCS))
-
-# Note that if the objcopy command starts getting too long, we can use `@file` to read
-# command-line options from `file` instead.
-define strip_implib
-	$(OBJCOPY) $(1).a $(STRIP_EXPORTED_FUNCS) $(1).a.stripped
-	mv $(1).a.stripped $(1).a
-endef
 endif
 
 $(build_shlibdir)/libjulia.$(JL_MAJOR_MINOR_SHLIB_EXT): $(LIB_OBJS) | $(build_shlibdir) $(build_libdir)
-	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@) $(LOADER_CFLAGS) -DLIBRARY_EXPORTS -shared $(SHIPFLAGS) $(LIB_OBJS) -o $@ $(JLIBLDFLAGS) $(LOADER_LDFLAGS) $(RPATH_LIB)) $(call SONAME_FLAGS,libjulia.$(JL_MAJOR_SHLIB_EXT))
-	$(INSTALL_NAME_CMD)libjulia.$(SHLIB_EXT) $@
+	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@) $(LOADER_CFLAGS) -DLIBRARY_EXPORTS -shared $(SHIPFLAGS) $(LIB_OBJS) -o $@ \
+		$(JLIBLDFLAGS) $(LOADER_LDFLAGS) $(RPATH_LIB) $(call SONAME_FLAGS,libjulia.$(JL_MAJOR_SHLIB_EXT)))
+	@$(INSTALL_NAME_CMD)libjulia.$(SHLIB_EXT) $@
+
+$(build_shlibdir)/libjulia-debug.$(JL_MAJOR_MINOR_SHLIB_EXT): $(LIB_DOBJS) | $(build_shlibdir) $(build_libdir)
+	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@) $(LOADER_CFLAGS) -DLIBRARY_EXPORTS -shared $(DEBUGFLAGS) $(LIB_DOBJS) -o $@ \
+		$(JLIBLDFLAGS) $(LOADER_LDFLAGS) $(RPATH_LIB) $(call SONAME_FLAGS,libjulia-debug.$(JL_MAJOR_SHLIB_EXT)))
+	@$(INSTALL_NAME_CMD)libjulia-debug.$(SHLIB_EXT) $@
+
+$(build_shlibdir)/libjulia.$(SHLIB_EXT) $(build_shlibdir)/libjulia-debug.$(SHLIB_EXT): $(build_shlibdir)/libjulia%.$(SHLIB_EXT): $(build_shlibdir)/libjulia%.$(JL_MAJOR_MINOR_SHLIB_EXT)
 ifneq ($(OS), WINNT)
-	@ln -sf $(notdir $@) $(build_shlibdir)/libjulia.$(JL_MAJOR_SHLIB_EXT)
-	@ln -sf $(notdir $@) $(build_shlibdir)/libjulia.$(SHLIB_EXT)
+	@$(call PRINT_LINK, ln -sf $(notdir $<) $@)
 else
-	$(call strip_implib,$(build_libdir)/libjulia.$(SHLIB_EXT))
+	@# Note that if the objcopy command starts getting too long, we can use `@file` to read
+	@# command-line options from `file` instead.
+	@$(call PRINT_LINK, $(OBJCOPY) $@.a $(STRIP_EXPORTED_FUNCS) $@.a.stripped)
+	@mv $@.a.stripped $@.a
```

Comment:
It makes me slightly uncomfortable that these commands are being tied to the `.$(SHLIB_EXT)` variants.  I think it would be better to have these as part of the source build so that we can't get into a situation where someone accidentally deletes only the `.$(JL_MAJOR_MINOR_SHLIB_EXT)` file and then rebuilds and the `$(OBJCOPY)` stuff never happens, because the link already exists.

Reply:
yeah, I was editing this part blind, and CI shows I need to fix it